src/utils/fs.ts

```diff
@@ -41,34 +41,36 @@ export const safeUnlink = async (path: string) => {
 // filenames within those directories, if at least one of the directories
 // exists. If not, an error is thrown.
 export const listFunctionsDirectories = async function (srcFolders: string[]) {
-  const filenamesByDirectory = await Promise.all(
-    srcFolders.map(async (srcFolder) => {
-      try {
-        const filenames = await listFunctionsDirectory(srcFolder)
+  const filenamesByDirectory = await Promise.allSettled(
+    srcFolders.map((srcFolder) => listFunctionsDirectory(srcFolder)),
+  )
+  const errorMessages: string[] = []
+  const validDirectories = filenamesByDirectory
+    .map((result) => {
+      if (result.status === 'rejected') {
+        if (result.reason instanceof Error && (result.reason as NodeJS.ErrnoException).code !== 'ENOENT') {
```

Comment:
Not a big deal, but I think this would be a bit easier to understand if we inverted the logic to treat `ENOENT` as the special case — i.e. if the error is `ENOENT` we return `null`, otherwise we throw. Might also be good to add a comment explaining the intent.